### DISCUSSION TITLE
[Feature] Rename and Move Tags

### DISCUSSION THREAD

**Discussion by MaroIshiku** [immich-app/immich#12210] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

It would be a great addition, if you would be able to simply rename tags and move them.
This would enable us to bulkmove photos/videos to a "new tag" without the need of deleting those tags on the pictures, and adding the new one.
Example 1 (Translating tag to english): Hintergrundbild > Wallpapers
Example 2 (Nesting the tag for better filters): Wallpapers 1080p > Wallpapers/1080p

In this usecase I currently need to remove the tag "Hintergrundbild" from every photo and need to add the tag "Wallpapers" or "Wallpapers/1080p" on 120 photos i took

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by benjamonnguyen** [comment]
I can take this if it’s something the maintainers want to add. 

Feature: enable updating tag names with hierarchy support

**Reply by bo0tzz** [reply]
Go for it!

**Reply by fabb** [reply]
Awesome!

**Reply by jtagcat** [reply]
PR was closed as stale https://github.com/immich-app/immich/pull/17902

**Reply by fabb** [reply]
Oh no.
Is there a safe way to edit xmp files of the internal lib and sync the changes? I‘ve done it for external libs, but i read it’s discouraged for the internal lib.

**Reply by bo0tzz** [reply]
Editing (and creating/removing) XMP files is perfectly fine, you just shouldn't touch the image files.

**Reply by fabb** [reply]
Ok thank you, that is a relief!

**Comment by Cedric-Boucher** [comment]
This would be a great addition and I would greatly appreciate it. My use case is that I have created tags for 50+ bird species as top-level tags in the hierarchy, but later realized I should've created a parent "birds" tag. There is currently no way to move all of these bird species tags to a different location in the hierarchy, and my only option right now is to create new tags and delete the old ones.

**Comment by petermgrund** [comment]
I'd really like to see this feature added soon. I'm hesitating to dive too deep into classifying my assets knowing that if I need to rename/ reorganize tags later (seems inevitable), I'll have to manually remove and reapply them.

**Comment by Noah-FTT** [comment]
This feature would be super helpful in conjunction with being able to bulk tag an upload at a time. Being able to edit tag parent groups after the fact would help our clean up and organization going forward a ton! Please!! Manually removing and re-adding tags after we realize it should have been a parent group is a struggle.

**Comment by speters** [comment]
I second this proposal.

Tagging is a first-class feature for photo management, so the tools for tagging should be good as well.

**Comment by jorbrock** [comment]
Hey all, I really wanted this feature implemented, so I took the awesome work by @benjamonnguyen and built on it more to create a version that works with immich 2.7.5. Would love some feedback on the open PR: https://github.com/immich-app/immich/pull/27909
